src/XAM.jl

```diff
@@ -1,29 +1,17 @@
 module XAM
 
+using BioGenerics
+import BioGenerics: isfilled #Note: used by `ismapped`.
+
 export
     SAM,
     BAM
 
-"""
-    flag(record::Union{SAM.Record, BAM.Record})::UInt16
-
-Get the bitwise flags of `record`. The returned value is a `UInt16` of each flag
-being OR'd together. The possible flags are:
+abstract type XAMRecord end
```

Comment:
Why not have `const XAMRecord = Union{BAMRecord, SAMRecord}`?

Reply:
Would a union block subtyping?